platform/device/builtin/db.go

```diff
@@ -54,19 +54,17 @@ func NewDB(db *bolt.DB, pubsubSvc pubsub.PublishSubscriber) (*DB, error) {
 }
 
 func (db *DB) List(opt device.ListDevicesOption) ([]device.Device, error) {
-	// TODO add filter/limit with ForEach
-	fmt.Println(opt.FilterSerial)
 	var devices []device.Device
 	err := db.View(func(tx *bolt.Tx) error {
 		b := tx.Bucket([]byte(DeviceBucket))
-		c := b.Cursor()
-		for k, v := c.First(); k != nil; k, v = c.Next() {
+		b.ForEach(func(k, v []byte) error {
 			var dev device.Device
 			if err := device.UnmarshalDevice(v, &dev); err != nil {
```

Comment:
Consider logging an error for this row instead. It could just be one bad unmarshalled record. For now though, this is probably fine.

Reply:
Actually, does ForEach continue on by itself?

Reply:
nope, you're right on this. https://github.com/boltdb/bolt/blob/fd01fc79c553a8e99d512a07e8e0c63d4a3ccfc5/tx.go#L124-L126